File: app/src/main/java/ua/hospes/nfs/marathon/utils/UiUtils.java
package ua.hospes.nfs.marathon.utils;

import android.app.Activity;
import android.view.View;

public class UiUtils {
    @SuppressWarnings("unchecked")
    public static <T> T findView(View root, int id) {
        if (root == null) return null;
        try {
            return (T) root.findViewById(id);
        } catch (ClassCastException e) {
            return null;
        }
    }

    @SuppressWarnings("unchecked")
    public static <T> T findView(View root, Object tag) {
        if (root == null) return null;
        try {
            return (T) root.findViewWithTag(tag);
        } catch (ClassCastException e) {
            return null;
        }
    }

    @SuppressWarnings("unchecked")
    public static <T> T findView(Activity root, int id) {
        if (root == null) return null;
        try {
            return (T) root.findViewById(id);
        } catch (ClassCastException e) {
            return null;
        }
    }
}

File: app/src/main/java/ua/hospes/rtm/utils/UiUtils.kt
package ua.hospes.rtm.utils

import android.app.Activity
import android.view.View

object UiUtils {
    fun <T> findView(root: View?, id: Int): T? {
        if (root == null) return null
        try {
            return root.findViewById<View>(id) as T
        } catch (e: ClassCastException) {
            return null
        }

    }

    fun <T> findView(root: View?, tag: Any): T? {
        if (root == null) return null
        try {
            return root.findViewWithTag<View>(tag) as T
        } catch (e: ClassCastException) {
            return null
        }

    }

    fun <T> findView(root: Activity?, id: Int): T? {
        if (root == null) return null
        try {
            return root.findViewById<View>(id) as T
        } catch (e: ClassCastException) {
            return null
        }

    }
}